### DISCUSSION TITLE
Domotics not reading Tasmota Counter

### DISCUSSION THREAD

**Discussion by kenmercer** [arendst/Tasmota#17463] [Q&A] [answered]
I'm trying to read the W-hr pulses from a smart meter into Domoticz using a nodeMCU (and a little hardware) configured as a Sonoff basic. DzSensorIdx4 is 32 and Tasmota correctly reads the pulses but the reading isn’t getting to Domoticz. 

However, from the terminal, I can inject a test value which correctly sets the Domoticz counter:
`mosquitto_pub -d -t 'domoticz/in' -m '{"idx":32,"svalue":"123"}'  -h 192.168.xx.xx -u xxxx -P xxxx`
giving (with mosquitto_sub):
`domoticz/in {"idx":32,"svalue":"123"}`

The actual mqtt message coming from Tasmota is:
`tele/tasmota_056835/SENSOR {"Time":"2022-12-21T01:27:10","COUNTER":{"C1":105}}`

This compares with a working temperature sensor:
`domoticz/in {"idx":25,"nvalue":0,"svalue":"23.3","Battery":100,"RSSI":8}`

It appears to me that the tele/tasmota message is a general message broadcast to anything listening (there’s one from the temperature sensor too) while domoticz/in along with idx: is required for the connection to work.

### Other info
Tasmota Version	12.3.1, Domoticz version: 2022.1
In Domoticz the power meter is idx:32
I’ve configured the nodeMCU to “Generic” with the same result.
I’ve looked at all the “SetOption” and Dz commands hoping to find a relivant configuration control.
And googled heaps!

A possible solution could be to trigger a rule which transmits the current counter value but surely there’s a better approach? What have I missed? Thanks in advance!

**Comment by arendst** [comment, marked as answer]
As you noticed you are using a counter. For Domoticz to work using the counter driver you need to select ``DzSensorIdx6`` as can be seen from the GUI.

**Reply by kenmercer** [reply]
Duh - I was sure I'd tried **all** the SensorIdx values **including** Idx6. Many thanks for this and your contributions over the years!
